packages/model-viewer/src/three-components/Renderer.ts

```diff
@@ -56,15 +59,27 @@ const DEFAULT_LAST_STEP = 3;
  * the texture.
  */
 export class Renderer extends EventDispatcher {
-  private static _singleton = new Renderer({debug: isDebugMode()});
+  private static _singleton = new Renderer(
+      {powerPreference: DEFAULT_POWER_PREFERENCE, debug: isDebugMode()});
```

Comment:
It turns out this is the only place we need to query for the setting. And setting the value ahead of time was the thing I needed help with. Turns out our docs have a typo (sort of). 

The docs say to set static properties before the library loads like so:
```
<script>
ModelViewerElement = self.ModelViewerElement || {};
ModelViewerElement.dracoDecoderLocation = 'http://example.com/location/of/draco/decoder/files/';
</script>
```

Which works, except when you use `<script type="module">`, which is now standard practice. It should say:

```
<script type="module">
self.ModelViewerElement = self.ModelViewerElement || {};
self.ModelViewerElement.dracoDecoderLocation = 'http://example.com/location/of/draco/decoder/files/';
</script>
```

Apparently in the olden days if you forgot to declare a variable (no `const`, `let`, or `var`) then it would just merrily attach it to the window, which is what `self` does properly. Modules won't let you get away with being that sloppy. 

So, since `self.ModelViewerElement` might have been declared by someone before we were loaded, we just need to query it to get the set value: `powerPreference: (self as any).ModelViewerElement?.powerPreference ||
        DEFAULT_POWER_PREFERENCE`

Reply:
Thanks for the thorough explanation